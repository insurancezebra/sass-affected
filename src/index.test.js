import sassAffected from "./index";

it("works with a node that is the root", async () => {
  expect(await sassAffected("mocks", ["mocks/rootA.scss"])).toMatchSnapshot();
});

it("works with a node imported by the root", async () => {
  expect(
    await sassAffected("mocks", ["mocks/intermediaryA.scss"])
  ).toMatchSnapshot();
});

it("works with a leaf node imported through an intermediary", async () => {
  expect(await sassAffected("mocks", ["mocks/leafA.scss"])).toMatchSnapshot();
});

describe("with multiple roots", () => {
  it("works with a node imported by the root", async () => {
    expect(
      await sassAffected("mocks", ["mocks/intermediaryB.scss"])
    ).toMatchSnapshot();
  });

  it("works with a leaf node imported through an intermediary", async () => {
    expect(await sassAffected("mocks", ["mocks/leafB.scss"])).toMatchSnapshot();
  });

  it("works with duplicate roots", async () => {
    expect(
      await sassAffected("mocks", [
        "mocks/intermediaryB.scss",
        "mocks/leafB.scss"
      ])
    ).toMatchSnapshot();
  });
});

describe("with mixed roots", () => {
  it("works with leaf nodes", async () => {
    expect(
      await sassAffected("mocks", ["mocks/leafA.scss", "mocks/leafB.scss"])
    ).toMatchSnapshot();
  });

  it("works with duplicate roots", async () => {
    expect(
      await sassAffected("mocks", [
        "mocks/leafA.scss",
        "mocks/leafB.scss",
        "mocks/intermediaryA.scss",
        "mocks/intermediaryB.scss"
      ])
    ).toMatchSnapshot();
  });
});

it("should display error when passing files that don't exist", async () => {
  const { missing, roots } = await sassAffected("mocks", ["mocks/leafC.scss"]);
  expect(missing.length).toBe(1);
  expect(roots.length).toBe(0);
});
