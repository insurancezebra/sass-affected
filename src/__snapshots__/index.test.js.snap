// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with mixed roots works with duplicate roots 1`] = `
Object {
  "missing": Array [],
  "roots": Array [
    Object {
      "file": "mocks/rootA.scss",
      "message": "There's a change in rootA.scss!",
    },
    Object {
      "file": "mocks/rootB.scss",
      "message": "There's a change in rootB.scss!",
    },
    Object {
      "file": "mocks/rootC.scss",
      "message": "There's a change in rootC.scss!",
    },
  ],
}
`;

exports[`with mixed roots works with leaf nodes 1`] = `
Object {
  "missing": Array [],
  "roots": Array [
    Object {
      "file": "mocks/rootA.scss",
      "message": "There's a change in rootA.scss!",
    },
    Object {
      "file": "mocks/rootB.scss",
      "message": "There's a change in rootB.scss!",
    },
    Object {
      "file": "mocks/rootC.scss",
      "message": "There's a change in rootC.scss!",
    },
  ],
}
`;

exports[`with multiple roots works with a leaf node imported through an intermediary 1`] = `
Object {
  "missing": Array [],
  "roots": Array [
    Object {
      "file": "mocks/rootB.scss",
      "message": "There's a change in rootB.scss!",
    },
    Object {
      "file": "mocks/rootC.scss",
      "message": "There's a change in rootC.scss!",
    },
  ],
}
`;

exports[`with multiple roots works with a node imported by the root 1`] = `
Object {
  "missing": Array [],
  "roots": Array [
    Object {
      "file": "mocks/rootB.scss",
      "message": "There's a change in rootB.scss!",
    },
    Object {
      "file": "mocks/rootC.scss",
      "message": "There's a change in rootC.scss!",
    },
  ],
}
`;

exports[`with multiple roots works with duplicate roots 1`] = `
Object {
  "missing": Array [],
  "roots": Array [
    Object {
      "file": "mocks/rootB.scss",
      "message": "There's a change in rootB.scss!",
    },
    Object {
      "file": "mocks/rootC.scss",
      "message": "There's a change in rootC.scss!",
    },
  ],
}
`;

exports[`works with a leaf node imported through an intermediary 1`] = `
Object {
  "missing": Array [],
  "roots": Array [
    Object {
      "file": "mocks/rootA.scss",
      "message": "There's a change in rootA.scss!",
    },
  ],
}
`;

exports[`works with a node imported by the root 1`] = `
Object {
  "missing": Array [],
  "roots": Array [
    Object {
      "file": "mocks/rootA.scss",
      "message": "There's a change in rootA.scss!",
    },
  ],
}
`;

exports[`works with a node that is the root 1`] = `
Object {
  "missing": Array [],
  "roots": Array [
    Object {
      "file": "mocks/rootA.scss",
      "message": "There's a change in rootA.scss!",
    },
  ],
}
`;
